networkx/algorithms/shortest_paths/astar.py

```diff
@@ -67,6 +67,14 @@ def astar_path(G, source, target, heuristic=None, weight="weight"):
     >>> print(nx.astar_path(G, (0, 0), (2, 2), heuristic=dist, weight="cost"))
     [(0, 0), (0, 1), (0, 2), (1, 2), (2, 2)]
 
+    Notes
+    -----
+    Edge weight attributes must be numerical.
+    Distances are calculated as sums of weighted edges traversed.
```

Comment:
The point I was getting at here with the original comment is that the "Distances are calculated as sums of weighted edges traversed" is not quite correct for the A* case, as the distance *also* takes into account the heuristic, right?

Reply:
If I understand correctly, the heuristic is an estimate of the distance. So the distance is still the sum of the weighted edges. But the estimate of the distance may include heuristic values. Perhaps we should add a sentence/phrase to make this clear?

Distances are calculated as sums of weighted edges traversed.
The algorithm uses heuristics to estimate these distances.

Reply:
> If I understand correctly, the heuristic is an estimate of the distance. So the distance is still the sum of the weighted edges. But the estimate of the distance may include heuristic values.

That was my understanding as well. As a concrete example of where confusion may arise:

```python
>>> G = nx.DiGraph()
>>> G.add_weighted_edges_from([(0, 1, 10), (1, 2, 10)])
>>> G.add_weighted_edges_from([(0, 2, 21)])
>>> nx.astar_path(G, 0, 2)  # No heuristic, weights only 
[0, 1, 2]
>>> heur = lambda u, v: 100 if (u, v) == (1, 2) else 0
>>> nx.astar_path(G, 0, 2, heuristic=heur)  # "distance" between 1, 2 is now much larger because of heuristic - weights unchanged
[0, 2]
```

Of course, there's nuance here. For example, A* is only guaranteed to work if the heuristic is "valid"; I doubt `heur` from the contrived example above counts.

> Perhaps we should add a sentence/phrase to make this clear?

Agreed, I like your wording suggestion :+1: . There are some other finer points re: astar and weights vs. heuristics vs. distances that I think are worth investigating further, but those are broader topics to be discussed elsewhere so they don't block this PR!